docs/operations/scalable-query-frontend.md

```diff
@@ -5,25 +5,52 @@ weight: 5
+For every query frontend the querier adds a [configurable number of goroutines](https://github.com/cortexproject/cortex/blob/1797adfed2979f6096c3305b0dc9162c1ec0c046/pkg/querier/worker/worker.go#L212)
+which are each capable of executing a query.
+Therefore, scaling the query frontend impacts the amount of work each individual querier is attempting to do at any given time.
 
-## Querier Max Concurrency
+Scaling up may cause a querier to attempt more work than they are configured for due to restrictions such as memory and cpu limits.
+Additionally, the PromQL engine itself is limited in the number of queries it can do as configured by the `-querier.max-concurrent` parameter.
+Attempting more queries concurrently than this value causes the queries to queue up in the querier itself.
 
-Make sure to configure the querier frontend worker to match max concurrency.  This will allow the operator to freely scale the frontend up and down without impacting the amount of work an individual querier is attempting to perform.  More details [here](https://cortexmetrics.io/docs/proposals/scalable-query-frontend/#dynamic-querier-concurrency).
+For similar reasons scaling down the query frontend may cause a querier to not use its allocated memory and cpu effectively.
+This will lower effective resource utilization.
+Also, because individual queriers will be doing less work, this may cause increased queueing in the query frontends.
 
-### Example Configuration
+### Querier Max Concurrency
 
-**CLI**
-```
--querier.worker-match-max-concurrent=true
-```
+To guarantee that querier doesn't receive more queries that it can handle at the same time, make sure to configure the querier to match its PromQL concurrency with number of connections.
+This can be done by using `-querier.worker-match-max-concurrent=true` option, or `match_max_concurrent: true` field in `frontend_worker` section of YAML config file.
+This allows the operator to freely scale the frontend or scheduler up and down without impacting the amount of work an individual querier is attempting to perform.
 
-**Config File**
-```yaml
-frontend_worker:
-  match_max_concurrent: true
-```
\ No newline at end of file
+### Query Scheduler
+
+Query scheduler is a service that moves the in-memory queue from query frontend to a separate component.
+This makes scaling query frontend easier, as it allows running multiple query frontends without increasing the number of queues.
+
+In order to use query scheduler, both query frontend and queriers must be configured with query scheduler address
+(using `-frontend.scheduler-address` and `-querier.scheduler-address` options respectively).
+
+Note that querier will only fetch queries from query frontend OR query scheduler, but not both.
+If configured with both `-querier.frontend-address` and `-querier.scheduler-address`, querier will only use scheduler address.
```

Comment:
We should add a config validation to make it fail if both addresses are specified. What do you think?